evaluator/evaluator2.py

```diff
@@ -0,0 +1,312 @@
+from .common import (SystemPlatform, VulnerabilityDB, Vulnerability, EvaluatorMessageTypes,
+                     EvaluatorException, VmaasErrorException,
+                     PROMETHEUS_PORT, CFG, EVAL_COUNT, INV_ID_NOT_FOUND, EVAL_TIME,
+                     UNKNOWN_MSG, MESSAGE_PARSE_ERROR, VMAAS_ERRORS_SKIP, EMPTY_DATA_SKIP)
+
+LOGGER = get_logger(__name__)
+
+
+class Evaluator:
+    """Main evaluator class"""
+
+    def __init__(self, db_pool: AsyncConnectionPool, loop: asyncio.BaseEventLoop):
+        """Constructor"""
+        self.db_pool: AsyncConnectionPool = db_pool
+        self.loop = loop
+        self.evaluator_logic = EvaluatorLogic(db_pool)
+
+        LOGGER.info("consuming evaluator topic: %s", CFG.evaluator_topics)
+        self.evaluator_topic = MQReader(CFG.evaluator_topics, group_id="evaluator-new")
+        self.payload_tracker = MQWriter(CFG.payload_tracker_topic)
+        self.remediations_results = MQWriter(CFG.remediation_updates_topic)
+        self.evaluator_results = MQWriter(CFG.evaluator_results_topic)
+
+    async def init(self):
+        """Async constructor"""
+        await self.evaluator_logic.init()
+
+    async def _lock_system(self, inventory_id: str, conn: AsyncConnection) -> SystemPlatform:
+        """Lock system for update and return its row"""
+        async with conn.cursor(row_factory=dict_row) as cur:
+            await cur.execute("""
+                SELECT id, rh_account_id, vmaas_json, rule_results
+                FROM system_platform
+                WHERE inventory_id = %s
+                AND when_deleted IS NULL
+                FOR UPDATE
+                LIMIT 1
+                """, (inventory_id,))
+            row = await cur.fetchone()
+        if row:
+            return SystemPlatform(row["id"], row["rh_account_id"], row["vmaas_json"], row["rule_results"])
+        return None
+
+    async def _load_db_system_vulnerabilities(self, system_platform: SystemPlatform, conn: AsyncConnection) -> Dict[str, VulnerabilityDB]:
+        """Make map from marked vulnerabilities in db for system"""
+        cve_map = {}
+        async with conn.cursor(row_factory=dict_row) as cur:
+            await cur.execute("""
+                SELECT sv.state, cm.cve, sv.id as sv_id, sv.rule_id, sv.when_mitigated, sv.mitigation_reason,
+                       sv.advisories, sv.advisory_available, sv.rule_hit_details,
+                       sv.remediation_type_id
+                FROM system_vulnerabilities_active AS sv
+                JOIN cve_metadata AS cm ON sv.cve_id = cm.id
+                LEFT JOIN insights_rule AS ir ON sv.rule_id = ir.id
+                WHERE sv.system_id = %s AND sv.rh_account_id = %s
+                """, (system_platform.id, system_platform.rh_account_id,))
+            for cve_system in await cur.fetchall():
+                cve_map[cve_system["cve"]] = VulnerabilityDB(
+                    VulnerabilityState(cve_system["state"]), cve_system["sv_id"],
```

Comment:
I guess `cve_system["state"]` value will be None for existing rows. Doesn't `VulnerabilityState(None)` fail?

Reply:
yep, good point, will fix

Reply:
This should be fixed, I left a fixme there to remove it in future